File: mouse.js
require('fs').createReadStream('/dev/input/event13').on('data', function(d) {
  console.log(d.length,
    'type', d.readUInt16LE(16),
    'code', d.readUInt16LE(18),
    'value', d.readInt32LE(20)
  );
});

File: mouse.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#include <linux/input.h>
#include <fcntl.h>

#define MOUSEFILE "/dev/input/event13"

int main()
{
    int fd;
    struct input_event ie;
printf("%zu\n", sizeof(struct timeval));
    if((fd = open(MOUSEFILE, O_RDONLY)) == -1) {
        perror("opening device");
        exit(EXIT_FAILURE);
    }
    printf("waiting for event\n");
    while(read(fd, &ie, sizeof(struct input_event))) {
        printf("time %ld.%06ld\ttype %d\tcode %d\tvalue %d\n",
               ie.time.tv_sec, ie.time.tv_usec, ie.type, ie.code, ie.value);

    }
    return 0;
}
